### PR TITLE
Feature/#11 searchbar

### PR DESCRIPTION
**Pull request by Dennis-coder**
Added a searchbar in the frontend and an endpoint in the backend which the frontend will call when searching.

### CODE REVIEW

File: backend/src/routers/gameRouter.ts [hunk 1]
```diff
@@ -53,6 +53,43 @@ const addGameSchema = z.object({
...
+ *    "id": "clgkri8kk0000przwvkvbyj95",
+ *    "name": "Game 1",
+ *    "description": "Game 1 description",
+ * 	  "platformName": "Steam",
+ *	  "releaseDate": "2023-04-13",
+ *	  "playtime": "60"
+ *   }
+ * ]
+ */
+ gameRouter.get('/search', async (req, res) => {
```
Comment (x183) [inline backend/src/routers/gameRouter.ts:77]: Shouldn't this be a post request?

Comment (Dennis-coder) [reply backend/src/routers/gameRouter.ts:77]: I don't think so since we only query data, we don't do any processing or modification on the server

**Review by Krysset — APPROVED** [review]
Looks good, I just had to fix a tiny typo in apiDoc

File: backend/src/routers/gameRouter.ts [hunk 2]
```diff
@@ -53,6 +57,45 @@ const addGameSchema = z.object({
 	playtime: z.number().int().min(1)
 });
 
+/**
+ * @api {get} /api/v1/games/search/:term Request Games
```
Comment (Supergamer1337) [inline backend/src/routers/gameRouter.ts:61]: This is no longer accurate, as the code has been updated to instead use `/api/v1/games/search` with a query parameter of term, like this `/api/v1/games/search?term=something`. The docs should be updated accordingly

**Review by Supergamer1337 — CHANGES_REQUESTED** [review]

**Review by Supergamer1337 — APPROVED** [review]
With the doc changes, it looks good

**Review by x183 — COMMENTED** [review]
Looks good

**Pull request merged**